### PR TITLE
TF-825 Download email as EML file

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#825 

## Resolved

- Chrome browser:

https://github.com/linagora/tmail-flutter/assets/80730648/67cdc2f5-a3d1-4119-b0b2-a00a52a612ad

### CODE REVIEW

File: lib/features/email/data/network/email_api.dart [hunk 1]
```diff
@@ -734,4 +741,65 @@ class EmailAPI with HandleSetErrorMixin {
       throw NotFoundEmailRecoveryActionException();
     }
   }
+
+  Future<void> downloadMessageAsEML(
```
Comment (hoangdat) [inline lib/features/email/data/network/email_api.dart:745]: should writing test for this method

**Comment by dab246** [comment]
lgtm

**Comment by dab246** [comment]
- Updated demo for all browser. Please review again @hoangdat

File: lib/features/email/data/network/email_api.dart (hunk 1 above)
Comment (dab246) [reply lib/features/email/data/network/email_api.dart:746]: Done. Please see below commit.

**Review by hoangdat — APPROVED** [review]

**Review by tddang-linagora — APPROVED** [review]

**Comment by dab246** [comment]
- [ ] Add progress loading as download attachment

**Comment by dab246** [comment]
> * [x]  Add progress loading as download attachment

Done. Please see PR description on top

**Review by hoangdat — APPROVED** [review]

**Pull request merged**